core/src/main/java/org/geysermc/geyser/entity/type/Entity.java

```diff
@@ -90,7 +92,8 @@
     protected String nametag = "";
     /* Metadata end */
 
-    protected LongOpenHashSet passengers = new LongOpenHashSet();
+    protected IntList passengers = new IntArrayList();
+    protected int vehicleId = -1;
```

Comment:
To go along with the above comment, we can probably merge this with session.getRidingVehicle.